app/src/main/java/com/tks/trendsmeme/di/NetworkModule.kt

```diff
@@ -0,0 +1,32 @@
+package com.tks.trendsmeme.di
+
+import com.tks.trendsmeme.datalayer.constants.Constants
+import com.tks.trendsmeme.datalayer.datasource.remote.AppApiService
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+ object NetworkModule {
+     @Provides
+     @Singleton
+     fun provideRetrofit(): Retrofit {
+         return Retrofit.Builder().
+         baseUrl(Constants.BASE_URL).
+         addConverterFactory(GsonConverterFactory
+             .create())
+             .build()
```

Comment:
**Suggestion:** Chain the method calls in `provideRetrofit()` more clearly to enhance readability and maintainability. This will make the code easier to understand and modify in the future. [maintainability, importance: 7]
```suggestion
     return Retrofit.Builder()
         .baseUrl(Constants.BASE_URL)
         .addConverterFactory(GsonConverterFactory.create())
         .build()
```